src/test/java/org/example/DeliveryToFunctionality.java

```diff
@@ -0,0 +1,127 @@
+package org.example;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+import java.time.Duration;
+import java.util.List;
+
+public class DeliveryToFunctionality {
+
+    public WebDriver webDriver;
+    public String poland = "Poland";
+    public String zip = "90210";
+
+    @BeforeMethod
+    public void setup() {
+        WebDriverManager.chromedriver().setup();
+        // System.setProperty("webdriver.chrome.driver", "E:\\it\\java\\web_driver_test\\src\\test\\resources\\webdriver\\chromedriver.exe");
+
+        this.webDriver = new ChromeDriver();
+        webDriver.get("https://www.amazon.com/");
+        webDriver.manage().window().maximize();
+    }
+
+    @AfterMethod
+    public void finish() {
+        webDriver.close();
+        webDriver.quit();
+    }
+
+    @Test
+    public void theDeliverIcon() {
+
+        //      searching a field for a post code
+        WebElement destination = webDriver.findElement(By.id("nav-global-location-popover-link"));
+        destination.click();
+
+        //      waiting for a drop down window
+        WebElement zipDropDown = new WebDriverWait(webDriver, Duration.ofSeconds(10))
+                .until(ExpectedConditions.visibilityOfElementLocated(By.id("GLUXZipUpdateInput")));
+
+        //      entering the post code
+        WebElement zipCode = webDriver.findElement(By.id("GLUXZipUpdateInput"));
+        zipCode.sendKeys(zip);
+
+        WebElement apply = webDriver.findElement(By.xpath("//*[@id=\"GLUXZipUpdate\"]/span"));
+        apply.click();
+
+        //  waiting for a new window with continue button
+        WebElement continueDropDown = new WebDriverWait(webDriver, Duration.ofSeconds(10))
+                .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("(//*[contains (@class, 'a-button-input')])[4]")));
+
+        WebElement pressContinue = webDriver.findElement(By.xpath("(//*[contains (@class, 'a-button-input')])[4]"));
+        pressContinue.click();
+
+        //  waiting for refresh page
+        boolean waitToReload = new WebDriverWait(webDriver, Duration.ofSeconds(10)).until(ExpectedConditions.stalenessOf(destination));
+
+        WebElement zipCodeInfo = new WebDriverWait(webDriver, Duration.ofSeconds(10))
+                .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[@id= 'nav-global-location-data-modal-action']//span[contains(@id, 'glow-ingress-line2')]")));
```

Comment:
Not critical, but it is better to make this line shorter. It will be more readable.
if you have an element with id, you can just get this element. I mean, you can make this xpath shorter like this: //span[contains(@id, 'glow-ingress-line2')]

Reply:
Same for other lines